test installer::tests::install_basic is flaky

ex https://github.com/denoland/deno/pull/5161/checks?check_run_id=658884530

```
2020-05-09T12:19:01.2164644Z failures:
2020-05-09T12:19:01.2164813Z 
2020-05-09T12:19:01.2165875Z ---- installer::tests::install_basic stdout ----
2020-05-09T12:19:01.2180309Z ##[error]thread 'installer::tests::install_basic' panicked at 'Install failed: Custom { kind: Other, error: "Existing installation found. Aborting (Use -f to overwrite)." }', cli/installer.rs:331:5
2020-05-09T12:19:01.2183392Z stack backtrace:
2020-05-09T12:19:01.2184325Z    0:     0x560697dba924 - backtrace::backtrace::libunwind::trace::h5ae3454ebe40b621
2020-05-09T12:19:01.2184821Z                                at /cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.44/src/backtrace/libunwind.rs:86
2020-05-09T12:19:01.2185423Z    1:     0x560697dba924 - backtrace::backtrace::trace_unsynchronized::hbf6f267a483fb2c0
2020-05-09T12:19:01.2185897Z                                at /cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.44/src/backtrace/mod.rs:66
2020-05-09T12:19:01.2186493Z    2:     0x560697dba924 - std::sys_common::backtrace::_print_fmt::h37755015d94b9085
2020-05-09T12:19:01.2186693Z                                at src/libstd/sys_common/backtrace.rs:78
2020-05-09T12:19:01.2187317Z    3:     0x560697dba924 - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::h0561c8cd3569b40d
2020-05-09T12:19:01.2187534Z                                at src/libstd/sys_common/backtrace.rs:59
2020-05-09T12:19:01.2188083Z    4:     0x560697de5f6c - core::fmt::write::h86a597a35d48a212
2020-05-09T12:19:01.2188273Z                                at src/libcore/fmt/mod.rs:1063
2020-05-09T12:19:01.2188823Z    5:     0x560695487695 - std::io::Write::write_fmt::h1fd8f7c3aaade85d
2020-05-09T12:19:01.2189049Z                                at /rustc/4fb7144ed159f94491249e86d5bbd033b5d60550/src/libstd/io/mod.rs:1426
2020-05-09T12:19:01.2189681Z    6:     0x560697dae181 - std::io::impls::<impl std::io::Write for alloc::boxed::Box<W>>::write_fmt::h11a95b4981a262a0
2020-05-09T12:19:01.2189889Z                                at src/libstd/io/impls.rs:156
2020-05-09T12:19:01.2190679Z    7:     0x560697dbd8b5 - std::sys_common::backtrace::_print::h6c60c5ee09af0233
2020-05-09T12:19:01.2190885Z                                at src/libstd/sys_common/backtrace.rs:62
2020-05-09T12:19:01.2191452Z    8:     0x560697dbd8b5 - std::sys_common::backtrace::print::hf9d1db01634f9b2a
2020-05-09T12:19:01.2191635Z                                at src/libstd/sys_common/backtrace.rs:49
2020-05-09T12:19:01.2192214Z    9:     0x560697dbd8b5 - std::panicking::default_hook::{{closure}}::h1db9298dd213ae95
2020-05-09T12:19:01.2192411Z                                at src/libstd/panicking.rs:204
2020-05-09T12:19:01.2192966Z   10:     0x560697dbd5ab - std::panicking::default_hook::h3272faf6cc91fd2e
2020-05-09T12:19:01.2193159Z                                at src/libstd/panicking.rs:221
2020-05-09T12:19:01.2193719Z   11:     0x560697dbdf12 - std::panicking::rust_panic_with_hook::hb976084785e50594
2020-05-09T12:19:01.2193921Z                                at src/libstd/panicking.rs:470
2020-05-09T12:19:01.2194261Z   12:     0x560697dbdafb - rust_begin_unwind
2020-05-09T12:19:01.2194442Z                                at src/libstd/panicking.rs:378
2020-05-09T12:19:01.2196446Z   13:     0x560697de4681 - core::panicking::panic_fmt::h45f7d6868edb5678
2020-05-09T12:19:01.2196642Z                                at src/libcore/panicking.rs:85
2020-05-09T12:19:01.2197251Z   14:     0x560697de44a3 - core::option::expect_none_failed::hd7ba247f271405e7
2020-05-09T12:19:01.2197447Z                                at src/libcore/option.rs:1211
2020-05-09T12:19:01.2198007Z   15:     0x5606960122cf - core::result::Result<T,E>::expect::h7f363f131825a7b0
2020-05-09T12:19:01.2198231Z                                at /rustc/4fb7144ed159f94491249e86d5bbd033b5d60550/src/libcore/result.rs:961
2020-05-09T12:19:01.2198811Z   16:     0x560694f6e3bf - deno::installer::tests::install_basic::h905212a4ffbbb655
2020-05-09T12:19:01.2199164Z                                at cli/installer.rs:331
2020-05-09T12:19:01.2199761Z   17:     0x560694f6e0ba - deno::installer::tests::install_basic::{{closure}}::h10812785d03ff243
2020-05-09T12:19:01.2199953Z                                at cli/installer.rs:319
2020-05-09T12:19:01.2200522Z   18:     0x560694d6e92e - core::ops::function::FnOnce::call_once::h439789176eb2464a
2020-05-09T12:19:01.2200796Z                                at /rustc/4fb7144ed159f94491249e86d5bbd033b5d60550/src/libcore/ops/function.rs:232
2020-05-09T12:19:01.2201441Z   19:     0x560695493b9f - <alloc::boxed::Box<F> as core::ops::function::FnOnce<A>>::call_once::h012be6ce828b5d0b
2020-05-09T12:19:01.2201672Z                                at /rustc/4fb7144ed159f94491249e86d5bbd033b5d60550/src/liballoc/boxed.rs:1017
2020-05-09T12:19:01.2202030Z   20:     0x560697dc9df7 - __rust_maybe_catch_panic
2020-05-09T12:19:01.2202217Z                                at src/libpanic_unwind/lib.rs:86
2020-05-09T12:19:01.2202768Z   21:     0x5606954aeffc - std::panicking::try::h4cc8d292e9fe5dd8
2020-05-09T12:19:01.2202988Z                                at /rustc/4fb7144ed159f94491249e86d5bbd033b5d60550/src/libstd/panicking.rs:281
2020-05-09T12:19:01.2203542Z   22:     0x5606954aeffc - std::panic::catch_unwind::h963079ac534210f7
2020-05-09T12:19:01.2203765Z                                at /rustc/4fb7144ed159f94491249e86d5bbd033b5d60550/src/libstd/panic.rs:394
2020-05-09T12:19:01.2204327Z   23:     0x5606954aeffc - test::run_test_in_process::h9a1821a43e552ab8
2020-05-09T12:19:01.2204518Z                                at src/libtest/lib.rs:542
2020-05-09T12:19:01.2205068Z   24:     0x5606954aeffc - test::run_test::run_test_inner::{{closure}}::h022f4c17b23ed2cb
2020-05-09T12:19:01.2205263Z                                at src/libtest/lib.rs:451
2020-05-09T12:19:01.2205857Z   25:     0x560695486dc6 - std::sys_common::backtrace::__rust_begin_short_backtrace::hc71773c677d1c1ee
2020-05-09T12:19:01.2206094Z                                at /rustc/4fb7144ed159f94491249e86d5bbd033b5d60550/src/libstd/sys_common/backtrace.rs:130
2020-05-09T12:19:01.2206807Z   26:     0x56069548b466 - std::thread::Builder::spawn_unchecked::{{closure}}::{{closure}}::hc99fda77faf31579
2020-05-09T12:19:01.2207022Z                                at /rustc/4fb7144ed159f94491249e86d5bbd033b5d60550/src/libstd/thread/mod.rs:475
2020-05-09T12:19:01.2207612Z   27:     0x56069548b466 - <std::panic::AssertUnwindSafe<F> as core::ops::function::FnOnce<()>>::call_once::he4800ebccddde5b2
2020-05-09T12:19:01.2207823Z                                at /rustc/4fb7144ed159f94491249e86d5bbd033b5d60550/src/libstd/panic.rs:318
2020-05-09T12:19:01.2208343Z   28:     0x56069548b466 - std::panicking::try::do_call::hc2cca8f4b20030d5
2020-05-09T12:19:01.2208548Z                                at /rustc/4fb7144ed159f94491249e86d5bbd033b5d60550/src/libstd/panicking.rs:303
2020-05-09T12:19:01.2208873Z   29:     0x560697dc9df7 - __rust_maybe_catch_panic
2020-05-09T12:19:01.2209054Z                                at src/libpanic_unwind/lib.rs:86
2020-05-09T12:19:01.2209556Z   30:     0x56069548bed6 - std::panicking::try::h061e25305e6e95c0
2020-05-09T12:19:01.2209735Z                                at /rustc/4fb7144ed159f94491249e86d5bbd033b5d60550/src/libstd/panicking.rs:281
2020-05-09T12:19:01.2210244Z   31:     0x56069548bed6 - std::panic::catch_unwind::he0ee09490dbb649b
2020-05-09T12:19:01.2210449Z                                at /rustc/4fb7144ed159f94491249e86d5bbd033b5d60550/src/libstd/panic.rs:394
2020-05-09T12:19:01.2211004Z   32:     0x56069548bed6 - std::thread::Builder::spawn_unchecked::{{closure}}::hab5579c66226575e
2020-05-09T12:19:01.2211218Z                                at /rustc/4fb7144ed159f94491249e86d5bbd033b5d60550/src/libstd/thread/mod.rs:474
2020-05-09T12:19:01.2211768Z   33:     0x56069548bed6 - core::ops::function::FnOnce::call_once{{vtable.shim}}::hbc145bf8f99ddd0f
2020-05-09T12:19:01.2211984Z                                at /rustc/4fb7144ed159f94491249e86d5bbd033b5d60550/src/libcore/ops/function.rs:232
2020-05-09T12:19:01.2212629Z   34:     0x560697da6f5f - <alloc::boxed::Box<F> as core::ops::function::FnOnce<A>>::call_once::h553ef812d1929d1b
2020-05-09T12:19:01.2212843Z                                at /rustc/4fb7144ed159f94491249e86d5bbd033b5d60550/src/liballoc/boxed.rs:1017
2020-05-09T12:19:01.2213465Z   35:     0x560697dc8eed - <alloc::boxed::Box<F> as core::ops::function::FnOnce<A>>::call_once::h51b51bce029ae491
2020-05-09T12:19:01.2213683Z                                at /rustc/4fb7144ed159f94491249e86d5bbd033b5d60550/src/liballoc/boxed.rs:1017
2020-05-09T12:19:01.2214227Z   36:     0x560697dc8eed - std::sys_common::thread::start_thread::hca943f45f04c8e46
2020-05-09T12:19:01.2214417Z                                at src/libstd/sys_common/thread.rs:13
2020-05-09T12:19:01.2214943Z   37:     0x560697dc8eed - std::sys::unix::thread::Thread::new::thread_start::h352e8a5875b189ee
2020-05-09T12:19:01.2215127Z                                at src/libstd/sys/unix/thread.rs:80
2020-05-09T12:19:01.2215614Z   38:     0x7fcc8e8cb6ba - start_thread
2020-05-09T12:19:01.2215927Z   39:     0x7fcc8e3e941d - clone
2020-05-09T12:19:01.2216240Z   40:                0x0 - <unknown>
2020-05-09T12:19:01.2216315Z 
2020-05-09T12:19:01.2216403Z 
2020-05-09T12:19:01.2216536Z failures:
2020-05-09T12:19:01.2216693Z     installer::tests::install_basic
2020-05-09T12:19:01.2216766Z 
2020-05-09T12:19:01.2216949Z test result: FAILED. 230 passed; 1 failed; 0 ignored; 0 measured; 0 filtered out
```

--- a/cli/installer.rs
+++ b/cli/installer.rs
@@ -80,8 +80,10 @@ deno {} "$@"
 }
 
 fn get_installer_root() -> Result<PathBuf, Error> {
-  if let Ok(env_dir) = env::var("DENO_INSTALL_ROOT").map(PathBuf::from) {
-    return env_dir.canonicalize();
+  if let Ok(env_dir) = env::var("DENO_INSTALL_ROOT") {
+    if !env_dir.is_empty() {
+      return PathBuf::from(env_dir).canonicalize();
+    }
   }
   // In Windows's Powershell $HOME environmental variable maybe null
   // if so use $USERPROFILE instead.
@@ -248,7 +250,12 @@ fn is_in_path(dir: &PathBuf) -> bool {
 #[cfg(test)]
 mod tests {
   use super::*;
+  use std::sync::Mutex;
   use tempfile::TempDir;
+
+  lazy_static! {
+    pub static ref ENV_LOCK: Mutex<()> = Mutex::new(());
+  }
 
   #[test]
   fn test_is_remote_url() {
@@ -317,6 +324,7 @@ mod tests {
 
   #[test]
   fn install_basic() {
+    let _guard = ENV_LOCK.lock().ok();
     let temp_dir = TempDir::new().expect("tempdir fail");
     let temp_dir_str = temp_dir.path().to_string_lossy().to_string();
     // NOTE: this test overrides environmental variables
@@ -325,8 +333,10 @@ mod tests {
     // It means that other test can override env vars when this test is running.
     let original_home = env::var_os("HOME");
     let original_user_profile = env::var_os("HOME");
+    let original_install_root = env::var_os("DENO_INSTALL_ROOT");
     env::set_var("HOME", &temp_dir_str);
     env::set_var("USERPROFILE", &temp_dir_str);
+    env::set_var("DENO_INSTALL_ROOT", "");
 
     install(
       Flags::default(),
@@ -356,6 +366,9 @@ mod tests {
     }
     if let Some(user_profile) = original_user_profile {
       env::set_var("USERPROFILE", user_profile);
+    }
+    if let Some(install_root) = original_install_root {
+      env::set_var("DENO_INSTALL_ROOT", install_root);
     }
   }
 
@@ -475,9 +488,11 @@ mod tests {
 
   #[test]
   fn install_custom_dir_env_var() {
+    let _guard = ENV_LOCK.lock().ok();
     let temp_dir = TempDir::new().expect("tempdir fail");
     let bin_dir = temp_dir.path().join("bin");
     std::fs::create_dir(&bin_dir).unwrap();
+    let original_install_root = env::var_os("DENO_INSTALL_ROOT");
     env::set_var("DENO_INSTALL_ROOT", temp_dir.path().to_path_buf());
 
     install(
@@ -499,6 +514,9 @@ mod tests {
     let content = fs::read_to_string(file_path).unwrap();
     assert!(content
       .contains(r#""run" "http://localhost:4545/cli/tests/echo_server.ts""#));
+    if let Some(install_root) = original_install_root {
+      env::set_var("DENO_INSTALL_ROOT", install_root);
+    }
   }
 
   #[test]

